packages/connection-form/src/components/connection-form.tsx

```diff
@@ -316,13 +316,27 @@ type ConnectionFormPropsWithoutSettings = {
   initialConnectionInfo: ConnectionInfo;
   connectionErrorMessage?: string | null;
   disableEditingConnectedConnection?: boolean;
-  onCancel?: () => void;
-  onConnectClicked?: (connectionInfo: ConnectionInfo) => void;
-  onSaveAndConnectClicked?: (connectionInfo: ConnectionInfo) => void;
-  onSaveClicked?: (connectionInfo: ConnectionInfo) => Promise<void>;
   onDisconnectClicked?: () => void;
   onAdvancedOptionsToggle?: (newState: boolean) => void;
   openSettingsModal?: (tab?: string) => void;
+
+  // form action buttons:
+
+  onCancel?: () => void;
+
+  // If onSaveClicked is specified, a Save button will show up. It will be the
```

Comment:
The comments are probably redundant now that the logic has been simplified and all buttons do what they say. But maybe they can guard against that regressing.